pkg/sql/lease_test.go

```diff
@@ -1048,3 +1048,125 @@ CREATE TABLE t.test (k CHAR PRIMARY KEY, v CHAR);
 		return nil
 	})
 }
+
+func (t *leaseTest) publishAndCheck(ctx context.Context, nodeID uint32, descID sqlbase.ID) {
+	desc, err := t.node(nodeID).Publish(ctx, descID, func(*sqlbase.TableDescriptor) error { return nil }, nil)
+	if err != nil {
+		t.Fatalf("error while publishing: %v", err)
+	}
+	table := desc.GetTable()
+
+	// Now check that the version that was updated is indeed observable
+	// in the database at the time it says it was modified.
+	//
+	// This checks that the modification timestamp is not lying about
+	// the transaction commit time (and that the txn commit time wasn't
+	// bumped past it).
+
+	txn := client.NewTxn(t.kvDB, roachpb.NodeID(nodeID), client.RootTxn)
+	// Make the txn look back at the known modification timestamp.
+	txn.SetFixedTimestamp(ctx, table.ModificationTime)
+
+	// Look up the descriptor.
+	descKey := sqlbase.MakeDescMetadataKey(descID)
+	dbDesc := &sqlbase.Descriptor{}
+	if err := txn.GetProto(ctx, descKey, dbDesc); err != nil {
+		t.Fatalf("error while reading proto: %v", err)
+	}
+	// Look at the descriptor that comes back from the database.
+	dbTable := dbDesc.GetTable()
+
+	if dbTable.Version != table.Version || dbTable.ModificationTime != table.ModificationTime {
+		t.Fatalf("db has version %d at ts %s, expected version %d at ts %s",
+			dbTable.Version, dbTable.ModificationTime, table.Version, table.ModificationTime)
+	}
+}
+
+func TestModificationTimeTxnOrdering(testingT *testing.T) {
+	testingT.Skip("#22479")
+
+	defer leaktest.AfterTest(testingT)()
+
+	// Decide how long we should run this.
+	maxTime := time.Duration(5) * time.Second
+	if testutils.NightlyStress() {
+		maxTime = time.Duration(2) * time.Minute
+	}
+
+	// Which table to exercise the test against.
+	const descID = keys.LeaseTableID
+
+	params, _ := tests.CreateTestServerParams()
+	t := newLeaseTest(testingT, params)
+	defer t.cleanup()
+
+	if _, err := t.db.Exec(`
+CREATE DATABASE t;
+CREATE TABLE t.test0 (k CHAR PRIMARY KEY, v CHAR);
+`); err != nil {
+		t.Fatal(err)
+	}
+
+	ctx := context.TODO()
+
+	// When to end the test.
+	end := time.Now().Add(maxTime)
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	log.Infof(ctx, "until %s", end)
+
+	go func() {
+		for count := 0; time.Now().Before(end); count++ {
+			log.Infof(ctx, "renaming test%d to test%d", count, count+1)
+			if _, err := t.db.Exec(fmt.Sprintf(`ALTER TABLE t.test%d RENAME TO t.test%d`, count, count+1)); err != nil {
+				t.Fatal(err)
+			}
+		}
+		wg.Done()
+	}()
+
+	go func() {
+		leaseMgr := t.node(1)
+		for time.Now().Before(end) {
+			log.Infof(ctx, "publishing new descriptor")
+			desc, err := leaseMgr.Publish(ctx, descID, func(*sqlbase.TableDescriptor) error { return nil }, nil)
```

Comment:
You have publishAndCheck() above. Why do you need this code?

Reply:
sorry the function above was a leftover from a previous attempt. cleaned up.